src/tools/cephfs_mirror/PeerReplayer.cc

```diff
@@ -516,8 +516,9 @@ int PeerReplayer::build_snap_map(const std::string &dir_root,
     uint64_t snap_id;
     if (is_remote) {
       if (!info.nr_snap_metadata) {
-        derr << ": snap_path=" << snap_path << " has invalid metadata in remote snapshot"
-             << dendl;
+        std::string invalid_metadata_str =  "snap_path=" + snap_path + " has invalid metadata in remote snapshot";
+        derr << ": " << invalid_metadata_str << dendl;
+        m_snap_sync_stats.at(dir_root).last_failed_reason = invalid_metadata_str;
```

Comment:
Does `last_failed_reason` get cleared (reset) ever?

Reply:
Yes, it gets cleared when the invalid metadata in the remote snapshot is cleared. I've verified that in vstart.